checkov/common/output/csv.py

```diff
@@ -100,16 +101,23 @@ def add_iac_resources(self, resource: Record | ExtraResource, git_org: str, git_
             # IaC resources shouldn't be added multiple times, if they don't have any misconfiguration
             return
 
-        self.iac_rows.append(
-            {
-                "Resource": resource.resource,
-                "Path": resource.file_path,
-                "git org": git_org,
-                "git repository": git_repository,
-                "Misconfigurations": misconfig,
-                "Severity": severity,
+        row = {
+            "Resource": resource.resource,
+            "Path": resource.file_path,
+            "git org": git_org,
+            "git repository": git_repository,
+            "Misconfigurations": misconfig,
+            "Severity": severity,
+        }
+
+        if len(resource.details):
+            self.iac_rows_have_details = True
+            row = {
+                **row,
+                "Details": "|".join(resource.details)
             }
```

Comment:
```suggestion
        if resource.details:
            self.iac_rows_have_details = True
            row["Details"] = "|".join(resource.details)
```
a non empty check can be done by just checking for truthiness of the value, like `if resource.details`
since the `row` dict is already created, you can just add an other key to it, no need for unpacking it again 🙂